pages/download.tsx

```diff
@@ -1,39 +1,244 @@
-import Button from "components/Button";
+import React, { useEffect, useState } from "react";
 import CodeBlockSection from "components/CodeBlockSection";
-import ExternalLink from "components/ExternalLink";
 import Head from "components/Head";
 import Layout from "components/Layout";
 import Section from "components/Section";
-import React from "react";
-import Link from "components/Link";
 
 import styles from "styles/pages/Download.module.scss";
+import PlatformDownloads from "components/PlatformDownloads";
+
+import linuxLogo from "assets/Tux_Mono.svg";
+import Tabs from "components/Tabs";
+import ExternalLink from "components/ExternalLink";
+import Button from "components/Button";
+import Loader from "components/Loader";
 
 type Props = {};
 
+type releaseObject = {
+  browser_download_url: string;
+  name: string;
+};
+
 const Download = (props: Props) => {
+  const [latestReleases, setLatestReleases] = useState<Array<releaseObject>>([]);
+  const [currentTab, setCurrentTab] = useState<string>();
+  const [primaryDownloadUrl, setPrimaryDownloadUrl] = useState<releaseObject>();
+  const [isGettingRelease, setIsGettingRelease] = useState(false);
+
+  useEffect(() => {
+    getLatestRelease();
+  }, []);
+
+  const getLatestBetaRelease = async (page: number = 1): Promise<any> => {
+    const cache = JSON.parse(localStorage.getItem("cachedRelease") ?? "{}");
+    if (cache.cacheExpires && parseInt(cache.cacheExpires) > new Date().getTime()) {
+      return cache.release;
+    }
+    const response = await (
+      await fetch(`https://api.github.com/repos/ferdium/ferdium-app/releases?per_page=${page}`)
+    ).json();
```

Comment:
Instead of using the fetch function, we could also use [Octokit ](https://octokit.github.io/rest.js/v18) and enjoy the typed returns for each endpoint

Reply:
Switched to Octokit :)